src/main/java/mod/chiselsandbits/core/api/ChiselAndBitsAPI.java

```diff
@@ -296,6 +297,19 @@ public void giveBitToPlayer(
 			}
 
 			ModUtil.feedPlayer( player.getEntityWorld(), player, ei );
+
+			final int entityItemState = ItemChiseledBit.getStackState( ei.getEntityItem() );
+			//entityItemState is always 0 when remote
+			if ( !player.getEntityWorld().isRemote && entityItemState != 0 )
+			{
+				if( ChiselsAndBits.getConfig().voidExcessBits )
+				{
+					if( !ItemChiseledBit.hasBitSpace( player, entityItemState ) )
+					{
+						player.addChatMessage( new TextComponentTranslation( "mod.chiselsandbits.result.void_excess" ) );
+					}
+				}
+			}
```

Comment:
Wouldn't this potentially make more sense inside the feedPlayer method, perhapes move feedPlayer out of ModUtil and into a new class that can track if its already informed the player for this particular result? That way this and the other invocations don't need this duplication?

Maybe something like,

BitInventoryFeeder feeder = new BitInventoryFeeder(player, world);
feeder.addItem(ei);

and then for the other one the addItem is inside the loop?